x-pack/plugins/security_solution/scripts/endpoint/resolver_generator_script.ts

```diff
@@ -181,33 +198,50 @@ async function main() {
     },
   }).argv;
   let ca: Buffer;
-  let kbnClient: KbnClient;
+
   let clientOptions: ClientOptions;
+  let url: string;
+  let node: string;
+  const toolingLogOptions = {
+    log: new ToolingLog({
+      level: 'info',
+      writeTo: process.stdout,
+    }),
+  };
+
+  let kbnClientOptions: KbnClientOptions = {
+    ...toolingLogOptions,
+    url: argv.kibana,
+  };
 
   if (argv.ssl) {
     ca = fs.readFileSync(CA_CERT_PATH);
-    const url = argv.kibana.replace('http:', 'https:');
-    const node = argv.node.replace('http:', 'https:');
-    kbnClient = new KbnClient({
-      log: new ToolingLog({
-        level: 'info',
-        writeTo: process.stdout,
-      }),
+    url = argv.kibana.replace('http:', 'https:');
+    node = argv.node.replace('http:', 'https:');
+    kbnClientOptions = {
+      ...kbnClientOptions,
       url,
       certificateAuthorities: [ca],
-    });
+    };
+
     clientOptions = { node, tls: { ca: [ca] } };
   } else {
-    kbnClient = new KbnClient({
-      log: new ToolingLog({
-        level: 'info',
-        writeTo: process.stdout,
-      }),
-      url: argv.kibana,
-    });
     clientOptions = { node: argv.node };
   }
-  const client = new Client(clientOptions);
+  let client = new Client(clientOptions);
+  // add endpoint user
+  await addUser(client);
```

Comment:
Should we remove the user at the end of the script? What happens if the user was created in an older script execution? Is it overwritten or is added twice?

Reply:
Fair question! I also thought about it and already working on it.

Reply:
addressed c21642a1703c592ef6e1dba51c771f4a41fe4541.

Reply:
addressed in 0f37f21fe49ea2169d183d77a4565b3adf4eb56f